### ISSUE TITLE
Break out the rspec helpers into a chef-rspec gem

### ISSUE THREAD

**Issue by lamont-granquist** [chef/cheffish#103]
Should probably live with chef/chef or follow the same release coordination as chef/chef and ohai will going forwards.

Having the rspec helpers in cheffish is weird as hell.  Even the README.md makes it clear that its weird "here's a pile of resources...  oh BTW here some 'bonus' rspec helpers...."  WTF?

Now cheffish is pulled into chef/chef because of the dep that chef/chef's integration tests have on cheffish's rspec helpers which is another tangled nightmare of tightly-coupled dependencies.

And most likely since there's apparently not a lot of unit testing around the rspec helpers (at least I was able to break the tests that use them in chef/chef without breaking any spec tests in this repo) that suggests that the compatibility between the rspec helpers and old versions of chef is most likely a lie -- certain its untested and at any point may become a lie.

**Comment by thommay** [comment]
yeah, let's pull it into a separate gem under chef/chef and version and pin the same way as for chef-config.
